lib/archethic/utils/regression/playbooks/smart_contract.ex

```diff
@@ -9,94 +9,136 @@ defmodule Archethic.Utils.Regression.Playbook.SmartContract do
-    amount_to_send = trunc(0.1 * @unit_uco)
+  def deploy(seed, data, host, port, storage_nonce_pubkey) do
+    Logger.debug("DEPLOY: Deploying contract")
 
-    contract_owner_address =
-      Crypto.derive_keypair("contract_playbook_seed", 0, :ed25519)
-      |> elem(0)
-      |> Crypto.derive_address()
+    secret_key = :crypto.strong_rand_bytes(32)
 
-    Logger.info(
-      "Genesis pool allocation owner is sending 10 UCO to #{Base.encode16(contract_owner_address)}"
-    )
+    # add the ownerships required for smart contract
+    data = %TransactionData{
+      data
+      | ownerships: [
+          %Ownership{
+            secret: Crypto.aes_encrypt(seed, secret_key),
+            authorized_keys: %{
+              storage_nonce_pubkey => Crypto.ec_encrypt(secret_key, storage_nonce_pubkey)
+            }
+          }
+          | data.ownerships
+        ]
+    }
+
+    {:ok, address} =
+      Playbook.send_transaction_with_await_replication(
+        seed,
+        :contract,
+        data,
+        host,
+        port
+      )
 
-    {:ok, funding_tx_address} = Playbook.send_funds_to(contract_owner_address, host, port)
-    Logger.info("Transaction address: #{Base.encode16(funding_tx_address)}")
+    Logger.debug("DEPLOY: Deployed at #{Base.encode16(address)}")
 
-    Process.sleep(1_000)
-    contract_balance = Playbook.get_uco_balance(contract_owner_address, host, port)
-    Logger.info("Contract got #{contract_balance} uco")
+    address
+  end
 
-    secret_key = :crypto.strong_rand_bytes(32)
+  def trigger(trigger_seed, contract_address, host, port) do
+    Logger.debug("TRIGGER: Sending trigger transaction")
 
-    {:ok, contract_tx_address} =
-      Playbook.send_transaction(
-        "contract_playbook_seed",
+    {:ok, trigger_address} =
+      Playbook.send_transaction_with_await_replication(
+        trigger_seed,
         :transfer,
         %TransactionData{
-          ownerships: [
-            %Ownership{
-              secret: Crypto.aes_encrypt("contract_playbook_seed", secret_key),
-              authorized_keys: %{
-                storage_node_public_key => Crypto.ec_encrypt(secret_key, storage_node_public_key)
-              }
-            }
-          ],
-          code: """
-          condition inherit: [
-            type: transfer,
-            uco_transfers: %{ "#{Base.encode16(recipient_address2)}" => #{amount_to_send} }
-          ]
-
-          actions triggered_by: interval, at: "* * * * * *" do
-            set_type transfer
-            add_uco_transfer to: "#{Base.encode16(recipient_address2)}", amount: #{amount_to_send}
-          end
-          """
+          recipients: [contract_address]
         },
         host,
         port
       )
 
-    Logger.info(
-      "Deployed smart contract at #{Base.encode16(contract_tx_address)} which sends 0.1 UCO to #{Base.encode16(recipient_address2)} each seconds"
-    )
+    # wait some time for the contract to trigger and create new tx
+    Process.sleep(250)
```

Comment:
Here you may request the last address of the SC chain and wait for it to change (with a max timeout). You can take inspiration from `Utils.await_confirmation`